lib/datadog/statsd/connection.rb

```diff
@@ -8,16 +8,6 @@ def initialize(telemetry: nil, logger: nil)
         @logger = logger
       end
 
-      # Close the underlying socket
-      def close
-        begin
-          @socket && @socket.close if instance_variable_defined?(:@socket)
-        rescue StandardError => boom
-          logger.error { "Statsd: #{boom.class} #{boom}" } if logger
-        end
-        @socket = nil
-      end
-
       def write(payload)
```

Comment:
Minor: Since subclasses need to provide `#close` and `#connect`, I would suggest making it more obvious by adding to this class something like:

```ruby
private

def connect
  raise 'Should be implemented by subclass'
end

def close
  raise 'Should be implemented by subclass'
end
```